lib/matplotlib/image.py

```diff
@@ -168,11 +169,34 @@ def _resample(
     allocating the output array and fetching the relevant properties from the
     Image object *image_obj*.
     """
+
+    # decide if we need to apply anti-aliasing if the data is upsampled:
+    # compare the number of displayed pixels to the number of
+    # the data pixels.
+    interpolation = image_obj.get_interpolation()
+    if interpolation == 'antialiased':
+        # don't antialias if upsampling by an integer number or
+        # if zooming in more than a factor of 3
+        # do antialiasing....
+        # compare the number of displayed pixels of the image to the number of
+        # the data pixels.
+        disppixelsX = int(transform.transform([data.shape[1], 0])[0])
+        disppixelsY = int(transform.transform([data.shape[0], 0])[0])
```

Comment:
shouldn't that be
```
dispy, dispx = transform.transform(data.shape).astype(int)
```
?
(the point being that the x and y transforms can be different)

Reply:
ooops